pkg/cluster/install.go

```diff
@@ -289,34 +289,58 @@ func setFieldCreatedByHive(createdByHive bool) database.OpenShiftClusterDocument
 }
 
 func (m *manager) bootstrap() []steps.Step {
-	s := []steps.Step{
+	s := []steps.Step{}
+
+	if m.doc.OpenShiftCluster.UsesWorkloadIdentity() {
+		s = append(s,
+			steps.Action(m.ensureClusterMsiCertificate),
+			steps.Action(m.initializeClusterMsiClients),
+		)
+	}
+
+	s = append(s,
 		steps.AuthorizationRetryingAction(m.fpAuthorizer, m.validateResources),
 		steps.Action(m.ensurePreconfiguredNSG),
 		steps.Action(m.ensureACRToken),
 		steps.Action(m.ensureInfraID),
 		steps.Action(m.ensureSSHKey),
 		steps.Action(m.ensureStorageSuffix),
 		steps.Action(m.populateMTUSize),
-
 		steps.Action(m.createDNS),
 		steps.Action(m.createOIDC),
-		steps.Action(m.initializeClusterSPClients), // must run before clusterSPObjectID
+	)
 
-		// TODO: this relies on an authorizer that isn't exposed in the manager
-		// struct, so we'll rebuild the fpAuthorizer and use the error catching
-		// to advance
-		steps.AuthorizationRetryingAction(m.fpAuthorizer, m.clusterSPObjectID),
+	if !m.doc.OpenShiftCluster.UsesWorkloadIdentity() {
+		s = append(s,
+			steps.Action(m.initializeClusterSPClients), // must run before clusterSPObjectID
+			// TODO: this relies on an authorizer that isn't exposed in the manager
+			// struct, so we'll rebuild the fpAuthorizer and use the error catching
+			// to advance
+			steps.AuthorizationRetryingAction(m.fpAuthorizer, m.clusterSPObjectID),
+		)
+	}
+
+	s = append(s,
 		steps.Action(m.ensureResourceGroup),
 		steps.Action(m.ensureServiceEndpoints),
 		steps.Action(m.setMasterSubnetPolicies),
 		steps.AuthorizationRetryingAction(m.fpAuthorizer, m.deployBaseResourceTemplate),
+	)
+
+	if m.doc.OpenShiftCluster.UsesWorkloadIdentity() {
+		s = append(s,
+			steps.Action(m.generateFederatedIdentityCredentials),
```

Comment:
Not sure if the `FederatedIdentityCredentials` ensure need to be part of this PR or not. But for update flow, the credentials will need to be ensure such that they exists and with correct values of issuer, audience etc.

Reply:
@cadenmarchese - can you confirm this? I didn't that in the [JIRA item](https://issues.redhat.com/browse/ARO-4375).

Reply:
@gouthamMN @rajdeepc2792 I am tempted to say that the update flow / ensure flow makes more sense under https://issues.redhat.com/browse/ARO-4371 rather than here. I have added it there. According to my understanding, we just need to ensure that federated credentials for all identities 1) exist 2) have correct issuer 3) have correct audience 4) have correct service account (?). That would cover update case as well as the create case.